src/Common/Admin/Help_Hub/Section_Builder.php

```diff
@@ -0,0 +1,241 @@
+ * structure and type safety, and stores built sections for later retrieval.
+ *
+ * @since   TBD
+ * @package TEC\Common\Admin\Help_Hub
+ */
+
+namespace TEC\Common\Admin\Help_Hub;
+
+/**
+ * Class Section_Builder
+ *
+ * Helper class to build and store consistent Help Hub sections with proper structure and type safety.
+ *
+ * @since   TBD
+ * @package TEC\Common\Admin\Help_Hub
+ */
+class Section_Builder {
+	/**
+	 * The section title.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $title;
+
+	/**
+	 * The section identifier/slug.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $slug;
+
+	/**
+	 * The section description.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $description = '';
+
+	/**
+	 * The section type.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $type = 'default';
+
+	/**
+	 * The section links or FAQ items.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $items = [];
+
+	/**
+	 * Static storage for all built sections.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string, array>
+	 */
+	private static array $sections = [];
+
+	/**
+	 * Create a new section instance.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $title The section title.
+	 * @param string $slug  The section identifier/slug.
+	 * @param string $type  Optional. The section type ('default' or 'faq').
+	 *
+	 * @return static
+	 */
+	public static function make( string $title, string $slug, string $type = 'default' ): self {
+		$instance        = new self();
+		$instance->title = $title;
+		$instance->slug  = $slug;
+		$instance->type  = $type;
+
+		return $instance;
+	}
+
+	/**
+	 * Set the section description.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $description The section description.
+	 *
+	 * @return $this
+	 */
+	public function set_description( string $description ): self {
+		$this->description = $description;
+
+		return $this;
+	}
+
+	/**
+	 * Add a link to the section (for default sections).
+	 *
+	 * @since TBD
+	 *
+	 * @param string $title The link title.
+	 * @param string $url   The link URL.
+	 * @param string $icon  Optional. The icon URL.
+	 *
+	 * @return $this
+	 */
+	public function add_link( string $title, string $url, string $icon = '' ): self {
+		if ( $this->type !== 'default' ) {
+			return $this;
+		}
+
+		$this->items[] = [
+			'title' => $title,
+			'url'   => $url,
+			'icon'  => $icon,
+		];
+
+		return $this;
+	}
+
+	/**
+	 * Add a FAQ item to the section (for FAQ sections).
+	 *
+	 * @since TBD
+	 *
+	 * @param string $question  The FAQ question.
+	 * @param string $answer    The FAQ answer.
+	 * @param string $link_text Optional. The "Learn More" link text.
+	 * @param string $link_url  Optional. The "Learn More" link URL.
+	 *
+	 * @return $this
+	 */
+	public function add_faq( string $question, string $answer, string $link_text = '', string $link_url = '' ): self {
+		if ( $this->type !== 'faq' ) {
```

Comment:
There is enough conditional stuff here about faq vs links I wonder if this should be an abstract or should unload most of this to distinct faq/link classes?